pkg/tangy/rpm.go

```diff
@@ -0,0 +1,93 @@
+package tangy
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+)
+
+const DefaultLimit = 500
+
+type RpmPackageSearch struct {
+	Name    string
+	Summary string
+}
+
+// RpmRepositoryVersionPackageSearch search for RPMs, by name, associated to repository hrefs, returning an amount up to limit
+func (t *tangyImpl) RpmRepositoryVersionPackageSearch(ctx context.Context, hrefs []string, search string, limit int) ([]RpmPackageSearch, error) {
+	conn, err := t.pool.Acquire(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Release()
+
+	if limit == 0 {
+		limit = DefaultLimit
+	}
+
+	repositoryIDs, versions, err := parseRepositoryVersionHrefs(hrefs)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing repository version hrefs: %w", err)
+	}
+
+	query := `SELECT DISTINCT ON (rp.name) rp.name, rp.summary
+              FROM rpm_package rp WHERE rp.content_ptr_id IN (`
+	for i := 0; i < len(repositoryIDs); i++ {
+		id := repositoryIDs[i]
+		ver := versions[i]
+
+		query += fmt.Sprintf(`
+			(
+    		    SELECT crc.content_id
+    		    FROM core_repositorycontent crc
+    		    INNER JOIN core_repositoryversion crv ON (crc.version_added_id = crv.pulp_id)
+    		    LEFT OUTER JOIN core_repositoryversion crv2 ON (crc.version_removed_id = crv2.pulp_id)
+    		    WHERE crv.repository_id = '%v' AND crv.number <= %v AND NOT (crv2.number <= %v AND crv2.number IS NOT NULL)
+				AND rp.name ILIKE CONCAT( '%%', '%v'::text, '%%')
+            )
+		`, id, ver, ver, search)
+
+		if i == len(repositoryIDs)-1 {
+			query += fmt.Sprintf(") ORDER BY rp.name ASC LIMIT %v;", limit)
+			break
+		}
+
+		query += "UNION"
+	}
+	rows, err := conn.Query(context.Background(), query)
```

Comment:
if you accidentally pass in an empty list, you end up getting a query error, it might be worth catching and just returning an empty list

Reply:
i think this is the last suggestion, otherwise looking good!